includes/class-llms-sessions.php

```diff
@@ -0,0 +1,416 @@
+	 *
+	 * @return void
+	 */
+	private function __construct() {
+
+		add_filter( 'cron_schedules', array( $this, 'add_cron_schedule' ) );
+
+		if ( ! wp_next_scheduled( 'llms_end_idle_sessions' ) ) {
+			wp_schedule_event( time(), 'every_five_mins', 'llms_end_idle_sessions' );
+		}
+		add_action( 'llms_end_idle_sessions', array( $this, 'end_idle_sessions' ) );
+
+	}
+
+	/**
+	 * Add cron schedule for session end interval checks.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $schedules Array of cron schedules
+	 * @return  array
+	 */
+	public function add_cron_schedule( $schedules ) {
+
+		// Adds every 5 minutes to the existing schedules.
+		$schedules['every_five_mins'] = array(
+			'interval' => MINUTE_IN_SECONDS * 5,
+			'display'  => sprintf( __( 'Every %d Minutes', 'lifterlms' ), 5 ),
+		);
+		return $schedules;
+
+	}
+
+	/**
+	 * End the 50 oldest idle sessions.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function end_idle_sessions() {
+
+		foreach ( $this->get_open_sessions() as $i => $event ) {
+			if ( $this->is_session_idle( $event ) ) {
+				$this->end( $event );
+			}
+		}
+
+	}
+
+	/**
+	 * End a session.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Event $start Event object for a session start.
+	 * @return LLMS_EVent
+	 */
+	protected function end( $start ) {
+
+		return LLMS()->events()->record(
+			array(
+				'actor_id'     => $start->get( 'actor_id' ),
+				'object_type'  => 'session',
+				'object_id'    => $start->get( 'object_id' ),
+				'event_type'   => 'session',
+				'event_action' => 'end',
+			)
+		);
+
+	}
+
+	/**
+	 * Ends the currently active session for the logged in user.
+	 *
+	 * @since [version]
+	 *
+	 * @return LLMS_Event|false
+	 */
+	public function end_current() {
+
+		$current = $this->get_current();
+		if ( ! $current ) {
+			return false;
+		}
+
+		return $this->end( $current );
+
+	}
+
+	/**
+	 * Retrieve the current session start event record for a given user.
+	 *
+	 * @since [version]
+	 *
+	 * @return LLMS_Event|false
+	 */
+	public function get_current() {
+
+		$user_id = get_current_user_id();
+		if ( ! $user_id ) {
+			return false;
+		}
+
+		$session = $this->get_last_session();
+		if ( ! $session ) {
+			return false;
+		}
+
+		$session = new LLMS_Event( $session->id );
+
+		if ( ! $this->is_session_open( $session ) ) {
+			return false;
+		}
+
+		return $session;
+
+	}
+
+	/**
+	 * Determine if a session is idle.
+	 *
+	 * A session is considered idle if it's open and no new events have been recorded
+	 * in the last 30 minutes.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Event $start Event record for the start of the session.
+	 * @return bool
+	 */
+	public function is_session_idle( $start ) {
+
+		// Session is closed so it can't be idle.
+		if ( ! $this->is_session_open( $start ) ) {
+			return false;
+		}
+
+		$now     = llms_current_time( 'timestamp' );
+		$timeout = apply_filters( 'llms_idle_session_timeout', 30 ) * MINUTE_IN_SECONDS;
```

Comment:
missing docs, that would clarify that we pass "minutes" hence justify why we have `* MINUTE_IN_SECONDS` outside the filter, so that I could avoid asking if that's intended or not <3